pkg/deploy-manager/subscription.go

```diff
@@ -360,3 +363,113 @@ func (t *DeployManager) waitForOCSOperator() error {
 
 	return nil
 }
+
+// UninstallOCS uninstalls ocs operator and storage clusters
+func (t *DeployManager) UninstallOCS(ocsRegistryImage string, localStorageRegistryImage string) error {
+	// Delete storage cluster and wait for it to be deleted
+	scs := &ocsv1.StorageClusterList{}
+	err := t.GetCrClient().List(context.TODO(), scs, client.InNamespace(InstallNamespace))
+	if err != nil {
+		return err
+	}
+
+	for _, sc := range scs.Items {
+		err = t.GetCrClient().Delete(context.TODO(), &sc)
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	lastReason := ""
+	timeout := 200 * time.Second
+	interval := 10 * time.Second
+	err = utilwait.PollImmediate(interval, timeout, func() (done bool, err error) {
+		err = t.GetCrClient().List(context.TODO(), scs, client.InNamespace(InstallNamespace))
+
+		if err != nil {
+			lastReason = fmt.Sprintf("Error talking to k8s apiserver: %v", err)
+			return false, nil
+		}
+
+		if len(scs.Items) > 0 {
+			lastReason = "Waiting on storagecluster to be deleted"
+			return false, nil
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("%v: %s", err, lastReason)
+	}
+
+	// Delete remaining operator manifests
+	co := t.generateClusterObjects(ocsRegistryImage, localStorageRegistryImage)
+	err = t.deleteClusterObjects(co)
+	if err != nil {
+		return err
+	}
+
+	// Delete all remaining deployments in the namespace
+	err = t.GetCrClient().DeleteAllOf(context.TODO(), &appsv1.Deployment{}, client.InNamespace(InstallNamespace))
+	if err != nil && client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	// Delete all remaining daemonsets in the namespace
+	err = t.GetCrClient().DeleteAllOf(context.TODO(), &appsv1.DaemonSet{}, client.InNamespace(InstallNamespace))
+	if err != nil && client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	// Delete all remaining pods in the namespace
+	err = t.GetCrClient().DeleteAllOf(context.TODO(), &k8sv1.Pod{}, client.InNamespace(InstallNamespace))
+	if err != nil && client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	// Delete PVCs in the namespace
+	opts := []client.DeleteAllOfOption {
+		client.InNamespace(InstallNamespace),
+		client.MatchingLabels(map[string]string{"nooba-core": "noobaa"}),
```

Comment:
i understand limiting the delete to the install namespace, but do we need this match label here for noobaa still?

Reply:
Since we are deleting the namespace, it's really not needed. Removed it.